### DISCUSSION TITLE
Cache dir cleanup?

### DISCUSSION THREAD

**Discussion by DizkoDan** [cupcakearmy/autorestic#260] [Q&A] [answered]
Hi,

I've been using autorestic for several months on a couple of servers, and it's been working out great! However, I happened to notice today that the /root/.cache/restic dir on one of my servers is well over 12G. Is there supposed to be a cleanup that automatically happens here, or am I expected to do something on my own? Reading in the restic docs, it seems like this should be automatic. I see that there is a `--cleanup-cache` flag for restic, but it isn't really clear how/when this should be used. My other server (while about 1/4 the size) is only using about 300MB for it's cache.

**Comment by DizkoDan** [comment, marked as answer]
So I looked into this a bit more, and it seems I was off base on the `cleanup-cache`. It's actually handled by the pruning schedule. I previously had it set to `--keep-monthly 3 --keep-weekly 4 --keep-last 5 --keep-daily 7 --keep-yearly 1`. I played with that a bit, and it removed 3 snapshots, and the cache size went down by about 1GB. Which doesn't really seem like much. 

At any rate, this seems like a restic issue rather than autorestic.

**Reply by cupcakearmy** [reply]
Thanks for the insight :)

**Comment by DizkoDan** [comment]
In case anyone runs into something like this, I found the problem. For some reason there was a lock on my remote repo, and I hadn't noticed it, because the cache command was showing the output from it just fine, and snapshots were still being created, But when I started poking at things to compare, I found it. Once I unlocked it, and looked at stats, I realized that there were hundreds of snapshots on the remote, that hadn't been pruned, with my daily job. Running the forget/prune against it, removed all the old, and the cache size dropped to 771MB which is completely reasonable.
